conformance/utils/tls/tls.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tls
+
+import (
+	"crypto/tls"
+	"fmt"
+	"testing"
+)
+
+type DialInfo struct {
+	Host       string
+	Port       string
+	ServerName string
+}
+
+type ExpectedOutput struct {
+	SANOrCommonName string
+}
+
+type TestCase struct {
+	DialInfo
+	ExpectedOutput
+}
+
+func InitiateTLSHandShakeAndValidateSNIMatch(t *testing.T, tlsDialInfo DialInfo, expectedOutput ExpectedOutput) {
```

Comment:
How will this function detect the failure? Methods like `t.Fail`, `T.Fatal` are not used.

Reply:
Sorry, It's my mistake. Thanks for pointing it out. I'll make changes and push the code.